### PR TITLE
feat(2346): Support for job level parameters

### PR DESCRIPTION
**Pull request by sagar1312**
## Context

Ability to define parameters at job level

## Objective

Create builds with job level parameters

## References

https://github.com/screwdriver-cd/screwdriver/issues/2346

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: lib/eventFactory.js [hunk 1]
```diff
@@ -473,53 +474,100 @@ function updateWorkflowGraph(config) {
...
- * @method updateEventParameters
+ * Determines the parameters for the build by compiling the list of parameters with default values from the
+ * `defaultParameters` and override the values if specified in `eventParameters`. Parameters from `eventParameters` are
+ * dropped if matching definition is not found in `defaultParameters`.
+ * @param {Object} defaultParameters Default Pipeline or Job parameter definitions
+ * @param {Object} eventParameters   Customized build parameters
+ * @resolves {Object}
+ */
+function validateAndMergeParameters(defaultParameters, eventParameters) {
+    const allowedParameters = Object.create(null);
```
Comment (tkyi) [inline lib/eventFactory.js:485]: wondering why we're using `Object.create(null)` vs `{}`, but I see that it was used in updateEventParameters() already

File: lib/eventFactory.js [hunk 2]
```diff
@@ -473,53 +474,100 @@ function updateWorkflowGraph(config) {
...
+                }
+
+                if (actualParameterValue !== undefined) {
+                    allowedParameters[name] = { value: actualParameterValue };
+                }
+            }
+        });
+    }
+
+    return allowedParameters;
```
Comment (tkyi) [inline lib/eventFactory.js:517]: will this return an Object vs null if `defaultParameters` does not exist?

File: lib/eventFactory.js (hunk 1 above)
Comment (jithine) [reply lib/eventFactory.js:485]: apparently quite interesting side effects https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/create#custom_and_null_objects

File: lib/eventFactory.js (hunk 2 above)
Comment (jithine) [reply lib/eventFactory.js:517]: it's an object, but a bit special https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/create#custom_and_null_objects

**Review by jithine — APPROVED** [review]

**Pull request merged**